store/postgres/src/relational.rs

```diff
@@ -518,21 +519,105 @@ impl Layout {
     pub fn find_range(
         &self,
         conn: &mut PgConnection,
-        entity_type: &EntityType,
+        entity_types: Vec<EntityType>,
+        causality_region: CausalityRegion,
         block_range: Range<BlockNumber>,
-    ) -> Result<BTreeMap<BlockNumber, Vec<Entity>>, StoreError> {
-        let table = self.table_for_entity(entity_type)?;
-        let mut entities: BTreeMap<BlockNumber, Vec<Entity>> = BTreeMap::new();
-        if let Some(vec) = FindRangeQuery::new(table.as_ref(), block_range)
-            .get_results::<EntityData>(conn)
+    ) -> Result<BTreeMap<BlockNumber, Vec<EntityWithType>>, StoreError> {
+        let mut tables = vec![];
+        let mut et_map: HashMap<String, Arc<EntityType>> = HashMap::new();
+        for et in entity_types {
+            tables.push(self.table_for_entity(&et)?.as_ref());
+            et_map.insert(et.to_string(), Arc::new(et));
+        }
+        let mut entities: BTreeMap<BlockNumber, Vec<EntityWithType>> = BTreeMap::new();
+        let lower_vec = FindRangeQuery::new(&tables, causality_region, false, block_range.clone())
```

Comment:
It would be helpful if there was a comment about what `lower_vec` and `upper_vec` represent